internal/services/controller/informers/informer.go

```diff
@@ -0,0 +1,42 @@
+package informers
+
+import (
+	"reflect"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"castai-agent/internal/services/controller/handlers"
+	"castai-agent/internal/services/controller/handlers/filters"
+	"castai-agent/internal/services/controller/handlers/transformers"
+	"castai-agent/internal/services/controller/handlers/transformers/cleaner"
+	"castai-agent/internal/services/controller/handlers/transformers/deletedfinalstateunknown"
+)
+
+type HandledInformer struct {
+	cache.SharedInformer
+	Handler handlers.Handler
+}
+
+var defaultTransformers = transformers.Transformers{
+	deletedfinalstateunknown.Transformer,
+	cleaner.Transformer,
+}
+
+func NewHandledInformer(
+	log logrus.FieldLogger,
+	queue workqueue.Interface,
+	informer cache.SharedInformer,
+	handledType reflect.Type,
+	filters ...filters.Filter,
+) *HandledInformer {
+	log = log.WithField("informer", handledType.String())
+	handler := handlers.NewHandler(log, queue, handledType, filters, defaultTransformers)
+	informer.AddEventHandler(handler)
+
+	return &HandledInformer{
+		SharedInformer: informer,
```

Comment:
nitpick: maybe SharedInformer - simply Informer?

Reply:
The SharedInformer is an embedded struct, renaming this to something else will cause more a.Informer.Method in other locations.